p2p/utils.py

```diff
@@ -33,6 +36,25 @@ def get_devp2p_cmd_id(msg: bytes) -> int:
     return rlp.decode(msg[:1], sedes=rlp.sedes.big_endian_int)
 
 
+@to_tuple
+def sequence_builder(start_number: int, max_length: int, skip: int, reverse: bool) -> Iterator[int]:
+    remaining_elements = max_length
+    next_element = start_number
+    step_size = skip + 1
+
+    while remaining_elements > 0:
+        if next_element < 0:
```

Comment:
This should also check that `next_element` is not greater than the maximum UINT256 size.